project/src/main/webapp/gmail.js

```diff
@@ -12,193 +12,206 @@
-                            const emailBodyEndIndex = (decodedEmail.substring(emailBodyStartIndex)).indexOf("Content-Type: text/html; charset=\"UTF-8\"")
-                            const emailBodyValue = decodedEmail.substring(emailBodyStartIndex, emailBodyStartIndex+emailBodyEndIndex);
-                            const emailDateStartIndex = decodedEmail.indexOf("Date:");
-                            const emailDateEndIndex = decodedEmail.substring(emailDateStartIndex).indexOf("Message-ID");
-                            var emailDateValue = decodedEmail.substring(emailDateStartIndex, emailDateStartIndex+emailDateEndIndex);
-                            if (emailDateValue.indexOf("X-Notifications") != -1) {
-                                emailDateValue = emailDateValue.substring(0, emailDateValue.indexOf("X-Notifications"));
-                            }
-                            var emailObject = {};
-                            emailObject[messageID] = {emailDate: emailDateValue, emailBody: emailBodyValue}; 
-                            if (emailBodyValue.indexOf("format=flowed") == -1 && emailDateValue.length > 1) {
-                                list[list.length] = emailObject;
+                        getPageOfMessages(request, result);
+                    } else {
+                        for (var i = 0; i < result.length; i++) {
+                            if (this.hasEmails(result)) {
+                                promiseArray.push(this.getMessage(result[i].id));
                             }
-                        });
-                    }
+                        }
+                        Promise.all(promiseArray).then(() => resolve()
+                        );
                     }
                 });
             };
-            console.log(list);
             var initialRequest = gapi.client.gmail.users.messages.list({
-            'userId': 'me',
-            'q': "When is the interview prep workshop?"
+                'userId': 'me',
+                'q': this.question
+            });
+            getPageOfMessages(initialRequest, []);
+            return Promise.all(promiseArray);
         });
-        getPageOfMessages(initialRequest, []);
-        const t1 = performance.now();
-        console.log(t1-t0);
+        return listMessagePromise;
     }
 
-    
-
-/**
- * Get Message with given ID.
- *
- * @param  {String} userId User's email address. The special value 'me'
- * can be used to indicate the authenticated user.
- * @param  {String} messageId ID of Message to get.
- * @param  {Function} callback Function to call when the request is complete.
- */
-function getMessage(userId, messageId, callback) {
-  var request = gapi.client.gmail.users.messages.get({
-    'userId': userId,
-    'id': messageId
-  });
-  request.execute(callback);
-}
\ No newline at end of file
+    /**  Retrieve messages using hardcoded queries and the signed-in email. */
+    listMessagesWithExtraction() {
```

Comment:
Should this function be removed? Seems like a duplicate for listMessages

Reply:
This was meant to be a protoype for the listMessages with the keyword extraction feature in case the email doesn't return anything. However, i don't know if it's really needed if we can cherry pick our questions for the demo. I can remove it for now, and then i can try to implement it later!